packages/demo/src/app/common/copy-to-clipboard.directive.ts

```diff
@@ -0,0 +1,17 @@
+import { Directive, ElementRef, HostListener } from '@angular/core';
+import { Clipboard } from '@angular/cdk/clipboard';
+
+@Directive({
+  selector: 'code[appCopyToClipboard]'
+})
+export class CopyToClipboardDirective {
+  constructor(
+    private readonly el: ElementRef<HTMLElement>,
+    private readonly clipboard: Clipboard
+  ) {}
+
+  @HostListener('click', ['$event']) private copyToClipboard(event: MouseEvent) {
+    event.preventDefault();
+    this.clipboard.copy(this.el.nativeElement.textContent);
```

Comment:
It's nice, thanks. Can you put a class on the element here that changes the icon to a checkmark (and disappears after a second or so)? That would be awesome.

Reply:
Just added that!